scripts/BUILD.bazel

```diff
@@ -0,0 +1,39 @@
+"""
+Kotlin-based scripts to help developers or perform continuous integration tasks.
+"""
+
+load("@io_bazel_rules_kotlin//kotlin:kotlin.bzl", "kt_jvm_binary", "kt_jvm_test")
+load("//scripts/src/java/org/oppia/android/scripts/regex:regex_pattern_assets.bzl", "retrieve_regex_pattern_check_assets")
+
+REGEX_PATTERN_CHECK_ASSETS = retrieve_regex_pattern_check_assets("//scripts/org/oppia/android/scripts/regex/assets")
+
+# Visibility for libraries that have binaries defined at this package, and libraries defined at
+# subpackage level.
+package_group(
+    name = "oppia_script_lib_visibility",
+    packages = [
+        "//scripts",
+        "//scripts/src/java/org/oppia/android/scripts/...",
+    ],
+)
+
+kt_jvm_binary(
+    name = "regex_pattern_validation_check",
+    testonly = True,
+    data = REGEX_PATTERN_CHECK_ASSETS,
+    main_class = "org.oppia.android.scripts.regex.RegexPatternValidationCheckKt",
+    runtime_deps = ["//scripts/src/java/org/oppia/android/scripts/regex:regex_pattern_validation_check_lib"],
+)
+
+kt_jvm_test(
+    name = "RegexPatternValidationCheckTest",
+    testonly = True,
+    srcs = ["src/test/org/oppia/android/scripts/regex/RegexPatternValidationCheckTest.kt"],
+    data = REGEX_PATTERN_CHECK_ASSETS,
+    deps = [
+        "//scripts/src/java/org/oppia/android/scripts/regex:regex_pattern_validation_check_lib",
+        "//testing:assertion_helpers",
+        "//third_party:com_google_truth_truth",
+        "//third_party:org_jetbrains_kotlin_kotlin-test-junit",
+    ],
+)
```

Comment:
@BenHenning 
I tried defining a macro in a new Bazel file from where we could directly retrieve the `REGEX_PATTERN_ASSETS`.
https://github.com/oppia/oppia-android/pull/3340/files#diff-3f5a0e252580a7a62a0de8d7b8ada4bc0c1885d1c4846dc06bc14139a9690cfaR7
If we define the `kt_jvm_test` in the root BUILD.bazel file of scripts then everything works fine. But if I try to define the test rule in the `regex` subpackage, then the bazel pipeline to process the textproto assets is not working. This is because it is not able to find an `assets` directory at the subpackage level. Therefore, I was left with no other choice but to define the test rule here. Please guide me on how can we achieve this in a clean way as per the best practices.
 
I also tried shifting the `assets` directory to the regex subpackage level, but that will not work for the root level binary rule, as it can't cross the subpackage boundary to reference the assets. Therefore, this also doesn't seems good.
Please share your thoughts!

Reply:
This is now resolved! as per our discussion on the `kt_jvm_library` approach.